utils/src/main/scala/za/co/absa/enceladus/utils/implicits/DataFrameImplicits.scala

```diff
@@ -68,6 +69,29 @@ object DataFrameImplicits {
       SparkUtils.withColumnIfDoesNotExist(df, colName, col)
     }
 
+    /**
+     * Set nullable property of column.
+     *
+     * @param columnName is the column name to change
+     * @param nullable   boolean flag to set the nullability of the column `columnName` to
+     */
+    def withNullableColumnState(columnName: String, nullable: Boolean): DataFrame = {
+      // Courtesy of https://stackoverflow.com/a/33195510
+
+      // modify [[StructField] with name `columnName` if its nullability differs
+      val newSchema = StructType(df.schema.map {
```

Comment:
**Minor:** I think a `foldLeft` would be able to signal back if change has happened avoiding the need for comparison right after.